File: backend/app/models/download_job.py
from beanie import Document, Indexed
from datetime import datetime, UTC
from enum import Enum
from pydantic import Field
from typing import Optional

class JobStatus(str, Enum):
    PENDING   = "pending"
    FETCHING  = "fetching"
    QUEUED    = "queued"
    RUNNING   = "running"
    MERGING   = "merging"
    DONE      = "done"
    FAILED    = "failed"
    CANCELLED = "cancelled"
    EXPIRED   = "expired"

class DownloadJob(Document):
    user_id: Indexed(str)
    url: str
    format: str

    quality: str = "best"
    codec: str = "any"
    hdr: str = "any"

    selected_format_id: Optional[str] = None
    format_string: Optional[str] = None

    status: JobStatus = JobStatus.PENDING
    progress: float = 0.0
    progress_text: str = ""
    title: Optional[str] = None
    thumbnail_url: Optional[str] = None
    file_path: Optional[str] = None
    file_name: Optional[str] = None
    file_size: Optional[int] = None
    error_message: Optional[str] = None

    created_at: datetime = Field(default_factory=lambda: datetime.now(UTC))
    updated_at: datetime = Field(default_factory=lambda: datetime.now(UTC))
    expires_at: Optional[datetime] = None

    class Settings:
        name = "download_jobs"
        indexes = ["user_id", "status", "created_at", [("expires_at", 1)]]

File: AyeVIdeoDownloader/Models/DownloadJob.swift
import Foundation

struct CreateDownloadRequest: Encodable {
    let url: String
    let format: String
    let quality: String?
    let codec: String?
    let hdr: String?
    let selected_format_id: String?
}

struct DownloadJobResponse: Decodable {
    let job_id: String
    let status: String
    let progress: Double
    let progress_text: String
    let title: String?
    let thumbnail_url: String?
    let file_size: Int?
    let file_name: String?
    let created_at: String
    let expires_at: String?
}
